packages/validation/frontend/sirius-components-validation/src/ValidationView.tsx

```diff
@@ -76,51 +50,39 @@ const useValidationViewStyle = makeStyles()((theme) => ({
   },
 }));
 
+const isValidationRefreshedEventPayload = (
+  payload: GQLValidationEventPayload | null
+): payload is GQLValidationRefreshedEventPayload =>
+  !!payload && payload.__typename === 'ValidationRefreshedEventPayload';
+
 export const ValidationView = ({ editingContextId }: WorkbenchViewComponentProps) => {
   const { classes } = useValidationViewStyle();
-  const [{ value, context }, dispatch] =
-    useMachine<StateMachine<ValidationViewContext, ValidationViewStateSchema, ValidationViewEvent>>(
-      validationViewMachine
-    );
-  const { toast, validationView } = value as SchemaValue;
-  const { id, validation, message } = context;
+  const { payload, complete } = useValidationViewSubscription(editingContextId);
 
-  const { error } = useSubscription<GQLValidationEventSubscription, GQLValidationEventVariables>(
-    validationEventSubscription,
-    {
-      variables: {
-        input: {
-          id,
-          editingContextId,
-        },
-      },
-      fetchPolicy: 'no-cache',
-      onData: ({ data }) => {
-        const handleDataEvent: HandleSubscriptionResultEvent = {
-          type: 'HANDLE_SUBSCRIPTION_RESULT',
-          result: data,
-        };
-        dispatch(handleDataEvent);
-      },
-      onComplete: () => {
-        const completeEvent: HandleCompleteEvent = { type: 'HANDLE_COMPLETE' };
-        dispatch(completeEvent);
-      },
-    }
+  const categories: Category[] = [];
+  const processedValidation: Validation = { categories };
+
+  let noDiagnostic: ReactNode = (
+    <div className={classes.idle}>
+      <Typography variant="subtitle2">No diagnostic available</Typography>
+    </div>
   );
 
-  useEffect(() => {
-    if (error) {
-      const { message } = error;
-      const showToastEvent: ShowToastEvent = { type: 'SHOW_TOAST', message };
-      dispatch(showToastEvent);
-    }
-  }, [error, dispatch]);
+  if (isValidationRefreshedEventPayload(payload) && !complete) {
```

Comment:
If we add another payload, the whole UI will disappear. You'll need to do like other representations and keep the latest refreshed event payload in the state.

Reply:
done